### PR TITLE
Fix lots_size

### PR DESCRIPTION
**Pull request by keisuke-umezawa**

### CODE REVIEW

**Comment by FelixKleinpeter** [comment]
Here is a release with unittest to verify the values of the lot sizes. I didn't applied any coefficient from market datas. I tried to applied a second ratio using the quote currency of the Trades, considering that the value of an unit of EURUSD is one euro. 

Results were very similar because offently quote currencies are very close (eg USD, EUR, GBP...). Depending on your advice I can merge it like that (removing comments) or using the quote currency ratio.

File: src/backlight/portfolio/strategy.py [hunk 1]
```diff
@@ -62,23 +64,37 @@ def equally_weighted_portfolio(
...
     for trade in trades:
         symbol = trade.symbol
         trade_currency = trade.currency_unit
 
-        ratio = 1
-        if trade_currency != currency_unit:
-            ratios = get_forex_ratios(mkt, trade_currency, currency_unit)
-            ratio = ratios.iloc[ratios.index.get_loc(trade.index[0]) - 1]
+        ratio = get_forex_ratio(trade.index[0], mkt, trade_currency, currency_unit)
+        #         if currency_unit == Currency[symbol[:3]]:
```
Comment (keisuke-umezawa) [inline src/backlight/portfolio/strategy.py]: Delete comments if they are not needed.

File: src/backlight/portfolio/strategy.py [hunk 2]
```diff
@@ -62,23 +64,37 @@ def equally_weighted_portfolio(
...
+        #         else:
+        #             ratio2 = get_forex_ratios(mkt, currency_unit,
+        #                     Currency[symbol[:3]]).loc[trade.index]
 
-        count_symbol = symbols.count(symbol)
-        principals[symbol] = principal / (nb_currencies * ratio * count_symbol)
-        lts[symbol] = int(principal / (max_amount))
+        principals[symbol] = principal / (nb_trades * ratio)
+        lot_sizes[symbol] = principal / (nb_trades * max_amount)
+    #         lot_sizes[symbol] = principal * ratio2 / (nb_trades * max_amount)
```
Comment (keisuke-umezawa) [inline src/backlight/portfolio/strategy.py]: Ditto.

File: tests/portfolio/test_strategy.py [hunk 3]
```diff
@@ -5,6 +5,11 @@
 
 import backlight
 from backlight.asset.currency import Currency
+from backlight.portfolio.strategy import (
+    equally_weighted_portfolio,
+    _calculate_principals_lot_sizes,
+)
```
Comment (keisuke-umezawa) [inline tests/portfolio/test_strategy.py]: Test target should be imported like

```
from backlight.portfolio import strategy as module
```

File: src/backlight/portfolio/strategy.py (hunk 2 above)
Comment (keisuke-umezawa) [reply src/backlight/portfolio/strategy.py]:
```
lot_sizes[symbol] = principal / (nb_trades * max_amount)
```

```
lot_sizes[symbol] = principal / (nb_trades * max_amount * ratio)
```

Comment (keisuke-umezawa) [reply src/backlight/portfolio/strategy.py]: lot_sizes[symbol] = principal * ratio2 / (nb_trades * max_amount)

**Comment by FelixKleinpeter** [comment]
I add new lots sizes computation from the second ratio. Because now lots sizes are just floats (not a dataframe), we compute the amount from ratio computed at t0, which is not accurate in my understanding, but difference is very small. Should I use dataframes for lots sizes or keep it as an float?

File: src/backlight/portfolio/strategy.py [hunk 4]
```diff
@@ -87,14 +87,14 @@ def _calculate_principals_lot_sizes(
         trade_currency = trade.currency_unit
 
         ratio = get_forex_ratio(trade.index[0], mkt, trade_currency, currency_unit)
-        #         if currency_unit == Currency[symbol[:3]]:
-        #             ratio2 = pd.Series(data=np.ones(trade.index.size), index=trade.index)
-        #         else:
-        #             ratio2 = get_forex_ratios(mkt, currency_unit,
-        #                     Currency[symbol[:3]]).loc[trade.index]
+        if currency_unit == Currency[symbol[:3]]:
```
Comment (keisuke-umezawa) [inline src/backlight/portfolio/strategy.py]: Do not use symbol[:3]. 

When symbol is `eurjpy` and currency_unit is `usd`,
ratio2 is exchange rate from `eur -> usd`.

You can calculate it by `eur -> jpy` x `jpy -> usd`.
Here, `rate` is `jpy -> usd` and `eur -> jpy` is `mkt.mid[trade.index[0]]`(= current price.).

1unit of eurjpy is `1eur`.
`1eur` -> `100jpy` -> `1usd`

Comment (keisuke-umezawa) [reply src/backlight/portfolio/strategy.py]: So, the lot size should be

```
lot_sizes[symbol] = principal * (current_price * rate) / (nb_trades * max_amount)
```

or 

```
lot_sizes[symbol] = principal / (current_price * rate) / (nb_trades * max_amount)
```

**Comment by FelixKleinpeter** [comment]
Currency grouped Portfolios with balanced lot sizes :

**Comment by keisuke-umezawa** [comment]
LGTM!

**Pull request merged**